t/lib/pgNode.pm

```diff
@@ -0,0 +1,171 @@
+# This program is open source, licensed under the PostgreSQL License.
+# For license terms, see the LICENSE file.
+#
+# Copyright (C) 2012-2021: Open PostgreSQL Monitoring Development Group
+
+
+=pod
+
+=head1 NAME
+
+pgNode - facet class extending PostgresNode for check_pgactivity tests
+
+=head1 DESCRIPTION
+
+This class should not be used directly to create objects. Its only aim is to
+extend the existing PostgresNode class, imported from PostgreSQL source code,
+without editing it so we can import new versions easily.
+
+See PostgresNode documentation for original methods.
+
+=cut
+
+package pgNode;
+
+use strict;
+use warnings;
+
+use Test::More;
+use Time::HiRes qw(usleep);
+use Cwd 'cwd';
+
+use PostgresNode;
+
+BEGIN {
+    # set environment vars
+    $ENV{'TESTDIR'}    = cwd;
+    delete $ENV{'PAGER'};
+
+    # Look for command 'true'
+    # It's under /bin on various Linux, under /usr/bin for macosx
+    if ( -x '/bin/true' ) {
+        $ENV{'PG_REGRESS'} = '/bin/true';
+    }
+    else {
+        $ENV{'PG_REGRESS'} = '/usr/bin/true';
+    }
+}
+
+sub new {
+    my $class = shift;
+    my $self = {};
+
+    $self->{'node'} = PostgresNode->get_new_node(@_);
+
+    bless $self, $class;
+
+    note('Node "', $self->{'node'}->name, '" uses version: ', $self->version);
+
+    return $self;
+}
+
+sub AUTOLOAD {
+    our $AUTOLOAD;
+    my $subname = $AUTOLOAD;
+    my $self    = shift;
+
+    $subname =~ s/^pgNode:://;
+
+    return $self->{'node'}->$subname(@_);
+}
+
+# Overload wait_for_catchup to pass the PostgresNode object as param
+sub wait_for_catchup {
+    my $self = shift;
+    my $stb = shift;
+
+    $self->{'node'}->wait_for_catchup($stb->{'node'}, @_);
+}
+
+=pod
+
+=head1 METHODS
+
+Bellow the changes and new methods implemented in this facet.
+
+=over
+
+=item $node->version()
+
+Return the PostgreSQL backend version.
+
+=cut
+
+sub version {
+    return $_[0]->{'node'}->{_pg_version};
+    #die "pgNode must not be used directly to create an object"
+}
+
+=item $node->switch_wal()
+
+Force WAL rotation.
+
+Return the old segment filename.
+
+=cut
+
+sub switch_wal {
+    my $self = shift;
+    my $result;
+
+    if ($self->version > '9.6') {
```

Comment:
Does this still use the overloaded <=> and cmp operators (from PostgresVersion.pm)?

Reply:
Yes. See second table there: https://perldoc.perl.org/overload#Magic-Autogeneration